src/query/execute_statement.c

```diff
@@ -17998,23 +18029,37 @@ do_create_server (PARSER_CONTEXT * parser, PT_NODE * statement)
     }
 
   /* PASSWORD */
-  if (si->pwd != NULL)
+  assert (si->pwd);
+  assert (si->pwd->node_type == PT_VALUE);
+  pwd = (char *) PT_VALUE_GET_BYTES (si->pwd);
+  if (pwd == NULL)
     {
-      assert (si->pwd->node_type == PT_VALUE);
-      attr_val[4] = (char *) PT_VALUE_GET_BYTES (si->pwd);
-      if (attr_val[4] == NULL)
+      error = ER_FAILED;
+      goto end;
+    }
+
+  error = get_dblink_password_decrypt (parser, pwd, &tmp_passwd);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+  error = get_dblink_password_encrypt (parser, (char *) db_get_string (&tmp_passwd), &passwd, false);
+  if (error != NO_ERROR)
+    {
+      if (!pt_has_error (parser))
 	{
-	  error = ER_FAILED;
-	  goto end;
+	  PT_ERRORf (parser, statement, "Failed to encryption password. error=%d", error);
```

Comment:
The error message id related to db link server, password encryption should be defined

Reply:
An error code has been added, but the error message-related processing will be done in the subsequent work.